aikau/src/main/resources/alfresco/lists/ItemSelectionMixin.js

```diff
@@ -0,0 +1,252 @@
+       * ignored.
+       * 
+       * @instance
+       * @param {object} payload The details of the selection request.
+       */
+      onItemSelection: function alfresco_lists_ItemSelectionMixin__onItemSelection(payload) {
+         if (payload)
+         {
+            if (payload.value === "selectAll")
+            {
+               // Select regardless of current status...
+               this.select();
+            }
+            else if (payload.value === "selectNone")
+            {
+               // De-select regardless of current status...
+               this.deselect();
+            }
+            else if (payload.value === "selectInvert")
+            {
+               // Invert the current status
+               this.onClick();
+            }
+            else if (payload.value === "selectFolders" && this.currentItem && this.currentItem.jsNode)
+            {
+               // Select if the current item is a container
+               if (this.currentItem.jsNode.isContainer)
+               {
+                  this.select();
+               }
+               else
+               {
+                  this.deselect();
+               }
+            }
+            else if (payload.value === "selectDocuments" && this.currentItem && this.currentItem.jsNode)
+            {
+               // Select if the current item is NOT a container
+               if (this.currentItem.jsNode.isContainer)
+               {
+                  this.deselect();
+               }
+               else
+               {
+                  this.select();
+               }
+            }
+            else if (payload.selectedItems)
+            {
+               // Check to see if the list of selected items contains the item that this instance has
+               // renderered.
+               array.some(payload.selectedItems, function(item) {
+                  var a = lang.getObject(this.itemKey, false, item);
+                  var b = lang.getObject(this.itemKey, false, this.currentItem);
+                  var match = ((a || a === 0) && a === b);
+                  if (match) {
+                     domClass.add(this.getSelectorNode(), "alfresco-lists-ItemSelectionMixin--checked");
+                  }
+                  return match;
+               }, this);
+            }
+         }
+      },
+      
+      /**
+       * Updates the CSS classes to indicate that the item has been selected and publishes 
+       * [the topic]{@link module:alfresco/documentlibrary/_AlfDocumentListTopicMixin#documentSelectedTopic}
+       * to indicate that an item has been selected.
+       * 
+       * @instance
+       * @fires module:alfresco/documentlibrary/_AlfDocumentListTopicMixin#documentSelectedTopic
+       */
+      select: function alfresco_lists_ItemSelectionMixin__select() {
+         domClass.add(this.getSelectorNode(), "alfresco-lists-ItemSelectionMixin--checked");
```

Comment:
This needs to be more semantic, and state should be applied as high as possible, in case other elements need to adjust their styling because of it.

```javascript
domClass.add(this.domNode, "alfresco-lists-ItemSelectionMixin--selected");
```